### PR TITLE
Migrate ?link to slash command

### PR DESCRIPTION
**Pull request by DonaldKellett**
Implement slash command `/link` with comparable functionality and remove `?link` command.

Related: #43

### CODE REVIEW

File: src/commands/link.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      .setName("topic")
+      .setDescription("The topic to link to")
+      .setRequired(true)
+      .addChoices(Object.keys(LINKS).map((k) => [k, k]))
+  )
+  .toJSON();
+
+export const call = async (interaction: CommandInteraction) => {
+  const topic = interaction.options.getString("topic", true);
+  await interaction.reply(`See <${LINKS[topic]}>`);
```
Comment (kazk) [inline src/commands/link.ts:34]: What do you think of hiding the url? [`hyperlink(description, url)`](https://discordjs.guide/popular-topics/builders.html#links) can be used. `LINKS` can map a short trigger keywords to url and a descriptive text. And that descriptive text can be used as a description for choices too, so it's clear what each means.

```
/link topic:katas

See [kata search page](https://www.codewars.com/kata)
```

Comment (DonaldKellett) [reply src/commands/link.ts:34]: Thanks, I'll look into it.

Another thing I noticed is that slash commands registered by our bot cannot be used as a reply to existing comments, so I'll probably have to add an extra parameter to have the bot mention a specific user when posting the link.

Comment (kazk) [reply src/commands/link.ts:34]: Is that a known limitation? Or is our bot missing some required scope?

Comment (DonaldKellett) [reply src/commands/link.ts]: Seems to me that it's a limitation of bot-registered slash commands. Anyway, I've added an optional parameter for directing a specific user to the selected link through mentions, let me know what you think.

File: src/commands/link.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  .addUserOption((option) =>
+    option.setName("target").setDescription("Direct the specified user to the given link")
+  )
+  .toJSON();
+
+export const call = async (interaction: CommandInteraction) => {
+  const topic = interaction.options.getString("topic", true);
+  const target = interaction.options.getUser("target");
+  const linkInfo = LINKS[topic];
+  let replyMsg = `See ${hyperlink(linkInfo.description, hideLinkEmbed(linkInfo.url))}`;
```
Comment (kazk) [inline src/commands/link.ts:75]: I didn't realize `hyperlink` adds link embed. I thought it was the same as writing `[name](url)`. The error message I added didn't expand: https://github.com/codewars/discord-bot/pull/41/files#r799916221

**Review by kazk — APPROVED** [review]
Looks good to me

File: src/commands/link.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// link
+export const data = new SlashCommandBuilder()
+  .setName("link")
+  .setDescription("Link to a given topic")
+  .addStringOption((option) =>
+    option
+      .setName("topic")
+      .setDescription("The topic to link to")
+      .setRequired(true)
+      .addChoices(Object.keys(LINKS).map((k) => [k, k]))
```
Comment (kazk) [inline src/commands/link.ts:64]: Was this (not using the description) intentional? Did it look weird?

Comment (DonaldKellett) [reply src/commands/link.ts:64]: Apologies, it wasn't intentional, I didn't quite understand what you meant. You mean that we can specify a description for each choice on top of a key-value pair?

Comment (kazk) [reply src/commands/link.ts:64]: I thought if we did
```typescript
  .addChoices(Object.keys(LINKS).map((k) => [LINKS[k].description, k]))
```
we can see the description when selecting choices.

Maybe I'm misremembering, but I thought it showed the "name" of the choice next to the "value".

Comment (DonaldKellett) [reply src/commands/link.ts:64]: Ah, I see. Now that I tried it, it seems that the "key" doesn't even have to be unique :man_facepalming: 

I'll remove duplicate entries and open a PR shortly.

**Pull request merged**